rpc/src/services/protocol/mod.rs

```diff
@@ -424,6 +424,47 @@ pub(crate) fn proto_get_contract_manager_key(
     }
 }
 
+pub(crate) fn proto_get_contract_balance(
+    _chain_id: &str,
+    block_id: &str,
+    pkh: &str,
+    persistent_storage: &PersistentStorage,
+    state: &RpcCollectedStateRef) -> Result<Option<String>, failure::Error> {
+
+    // get protocol and constants
+    let context_proto_params = get_context_protocol_params(
+        block_id,
+        None,
+        persistent_storage,
+        state,
+    )?;
+
+    let context = TezedgeContext::new(BlockStorage::new(&persistent_storage), persistent_storage.merkle());
```

Comment:
@adonagy 
there is a shared reference for TezedgeContext in rpc module,
please use that instead of creating new one here,
and also please check another usages in this file

Reply:
I see, cool, will change the usages.